### PR TITLE
docs: add missing method documentation for the types module 

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Makes the `ape.types` method-docs more usable.

**The will fix some of the broken links in the docs**

### How I did it

* Add missing docs
* Organize docs

Note, this module in general seems to need some work. I purposely did not document some of these methods / attributes.

### How to verify it

Build docs and check it out.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: src/ape/types/contract.py [hunk 1]
```diff
@@ -39,41 +45,93 @@ def __repr__(self) -> str:
...
+        Create this class from a dictionary of its properties.
+        The dictionary keys must be the same name as the properties.
+
+        Args:
+            params (dict): A dictionary of properties.
+
+        Returns:
+            :class:`~ape.types.contract.Bytecode`
+        """
+
```
Comment (evanblank3) [inline src/ape/types/contract.py:58]: I go back and forth about whether the dictionary methods should have a small example 🤷‍♂️

File: src/ape/types/contract.py [hunk 2]
```diff
@@ -39,41 +45,93 @@ def __repr__(self) -> str:
...
         return cls(**params)  # type: ignore
 
 
 class ContractInstance(SerializableType):
+    """
+    Serializable data representing a contract instance.
+    """
+
     contractType: str
+    """The name of the type of the smart-contract."""
```
Comment (evanblank3) [inline src/ape/types/contract.py:71]: The name of the type of smart-contract.

File: src/ape/types/contract.py [hunk 3]
```diff
@@ -39,41 +45,93 @@ def __repr__(self) -> str:
...
     settings: Optional[str] = None
     contractTypes: Optional[List[str]] = None
 
 
 class ABIType(SerializableType):
-    name: str = ""  # NOTE: Tuples don't have names by default
+    name: str = ""
     indexed: Optional[bool] = None
     type: Union[str, "ABIType"]
+    """The type, as defined in the contract, such as ``int``."""
```
Comment (evanblank3) [inline src/ape/types/contract.py:120]: Maybe more than on example value?

File: src/ape/types/contract.py [hunk 4]
```diff
@@ -39,41 +45,93 @@ def __repr__(self) -> str:
...
+
     internalType: Optional[str] = None
+    """The real type used in the byte-code, such as ``uint256``."""
```
Comment (evanblank3) [inline src/ape/types/contract.py:123]: Same as above

File: src/ape/types/contract.py [hunk 5]
```diff
@@ -221,23 +375,40 @@ def from_dict(cls, params: Dict):
...
 
 class Source(SerializableType):
+    """
+    A serializable type representing source material.
+    """
+
     checksum: Optional[Checksum] = None
-    urls: List[str] = []
+    """
+    The checksum of the source. Set by calling method
```
Comment (evanblank3) [inline src/ape/types/contract.py:396]: Set by calling the method

File: src/ape/types/contract.py [hunk 6]
```diff
@@ -221,23 +375,40 @@ def from_dict(cls, params: Dict):
...
+    :meth:`~ape.types.contract.Source.compute_checksum`.
+    """
+
     content: Optional[str] = None
     # TODO This was probably done for solidity, needs files cached to disk for compiling
     # If processing a local project, code already exists, so no issue
     # If processing remote project, cache them in ape project data folder
     installPath: Optional[str] = None
     type: Optional[str] = None
+    """The type of the source, such as ``"interface"``."""
```
Comment (evanblank3) [inline src/ape/types/contract.py:406]: The type of source,

**Review by evanblank3 — DISMISSED** [review]
All nitpicks

File: src/ape/types/contract.py (hunk 1 above)
Comment (unparalleled-js) [reply src/ape/types/contract.py:58]: hmm maybe we shouldn't document these method actually... I don't think users will call them too much directly

**Review by Ninjagod1251 — APPROVED** [review]

**Review by evanblank3 — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**